### ISSUE TITLE
Intercepted Protocols follow redirects, but chromium is unaware of those 

### ISSUE THREAD

**Issue by wjch-krl** [electron/electron#23741]
### Preflight Checklist

* [x] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/master/CONTRIBUTING.md) for this project.
* [x] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/master/CODE_OF_CONDUCT.md) that this project adheres to.
* [x] I have searched the issue tracker for an issue that matches the one I want to file, without success.

### Issue Details
When protocol's interceptor (created using [protocol.interceptStreamProtocol(scheme, handler[, completion])](https://www.electronjs.org/docs/api/protocol#protocolinterceptstreamprotocolscheme-handler-completion)) handles a redirect (with a 30X HTTP status code, and the 'Location' header set with a new resource location), it is being followed (resources are being downloaded from a new URL), however, Chromium is not-aware of the redirect (e.g. window.location is not updated). This causes invalid handling of requests with relative paths on the redirected page. For example: 

1. Navigate to https://google.com/ 
2. Server will send a redirect to https://www.google.com/
3. Electron will download https://www.google.com/
4. The site (https://www.google.com/), requests resources (images, style sheets etc) with a relative path (e.g. `<img alt="Google" id="hplogo" src="/images/branding/googlelogo/2x/googlelogo_color_272x92dp.png" style="padding-top:109px" data-atf="1" width="272" height="92">`
5. Electron tries to get the resource from https://google.com/images/branding/googlelogo/2x/googlelogo_color_272x92dp.png instead of https://www.google.com/images/branding/googlelogo/2x/googlelogo_color_272x92dp.png
6. The request fails with 404 

* **Electron Version:**
  * Found on 7.2.1, tested and confirmed on 9.0.0, and latest from master
* **Operating System:**
  * Found on Windows, 
* **Last Known Working Electron version:**
  * 4.2.9 (should be working before https://github.com/electron/electron/pull/18464/files)

### Expected Behavior
The intercepted protocol should follow Redirects, and chromium should be aware of those (e.g. window.location should be updated)

### Actual Behavior
The intercepted protocol follows Redirect, but chromium isn't aware of those (resulting in broken handling of sub-requests) 

### To Reproduce

```
git clone https://github.com/wjch-krl/electron-redirects
cd electron-redirects
npm install
npm run start
```
It will start a new application, that would navigate to http://google.com/, with protocol interceptor

### Screenshots

![image](https://user-images.githubusercontent.com/6430297/82799806-ef86a980-9e7a-11ea-95ce-a6cf0699dee9.png)

**Comment by wjch-krl** [comment]
I have prepared a PR with a fix for the issue:
https://github.com/electron/electron/pull/23742

**Comment by wjch-krl** [comment]
I have also backported the fix for 7.2.* tree. Will it be possible to merge it and release new 7.2. version? (it is important for us as we are using 7.2.1 currently, and we don't want to jump too many releases) - https://github.com/electron/electron/pull/23938 
@zcbenz what do you think?

**Comment by wjch-krl** [comment]
When can we expect a release from 7-3-x tree?
